compiler/pkg/generator/template/client.go.tmpl

```diff
@@ -146,47 +146,69 @@ func new{{$group.GroupTypeName}}(client *Clientset) *{{$group.GroupTypeName}} {
+			// return nil, err
+		}else{
+			log.Debugf("[Get{{$node.BaseNodeName}}ByName] Object: %s exists in cache", hashedName)
+			log.Infof("{{$node.BaseNodeName}}: %s from cache(informer), GenerationId: %+v",  result.DisplayName(), result.Generation)
+			log.Infof("{{$node.BaseNodeName}}: %s from cache(informer), Spec: %+v", result.DisplayName(), result.Spec)
+			result, _ := item.(*{{$node.GroupBaseImport}})
+			cache_obj = &{{$node.SimpleGroupTypeName}}{{$node.BaseNodeName}}{
+				client: group.client,
+				{{$node.BaseNodeName}}: result,
+			}
 		}
-
-		log.Debugf("[Get{{$node.BaseNodeName}}ByName] Object: %s exists in cache", hashedName)
-		result, _ := item.(*{{$node.GroupBaseImport}})
-		return &{{$node.SimpleGroupTypeName}}{{$node.BaseNodeName}}{
-			client: group.client,
-			{{$node.BaseNodeName}}: result,
-		}, nil
-	}else{
-		retryCount := 0
-		for {
-			result, err := group.client.baseClient.
-			{{$node.GroupTypeName}}().
-			{{$node.GroupResourceNameTitle}}().Get(ctx, hashedName, metav1.GetOptions{})
-			if err != nil {
-				log.Errorf("Failed to Get {{$node.GroupResourceNameTitle}}: %+v",err)
-				if errors.IsTimeout(err) || customerrors.Is(err, context.DeadlineExceeded) {
-					log.Errorf("[Retry Count: %d ] %+v",retryCount,err)
-					if retryCount == maxRetryCount {
-						log.Error("Max Retry exceed on Get {{$node.GroupResourceNameTitle}}")
-						return nil, err
-					}
-					retryCount +=1
-					time.Sleep(sleepTime * time.Second)
-				}else {
-					log.Errorf("[Get{{$node.BaseNodeName}}ByName]: %+v",err)
-					return nil, err
+	}
+	retryCount := 0
+	for {
+		result, err := group.client.baseClient.
+		{{$node.GroupTypeName}}().
+		{{$node.GroupResourceNameTitle}}().Get(ctx, hashedName, metav1.GetOptions{})
+		if err != nil {
+			log.Errorf("Failed to Get {{$node.GroupResourceNameTitle}}: %+v",err)
+			if errors.IsTimeout(err) || customerrors.Is(err, context.DeadlineExceeded) {
+				log.Errorf("[Retry Count: %d ] %+v",retryCount,err)
+				if retryCount == maxRetryCount {
+					log.Error("Max Retry exceed on Get {{$node.GroupResourceNameTitle}}")
+					// return nil, err
+					return_error=err
+					break
 				}
+				retryCount +=1
+				time.Sleep(sleepTime * time.Second)
+				continue
+			}else {
+				log.Errorf("[Get{{$node.BaseNodeName}}ByName]: %+v",err)
+				// return nil, err
+				return_error=err
+				break
 			}
-			return &{{$node.SimpleGroupTypeName}}{{$node.BaseNodeName}}{
-				client: group.client,
-				{{$node.BaseNodeName}}: result,
-			}, nil
 		}
+		log.Infof("{{$node.BaseNodeName}}: %s from apiserver, GenerationId: %+v",  result.DisplayName(), result.Generation)
+		log.Infof("{{$node.BaseNodeName}}: %s from apiserver, Spec: %+v", result.DisplayName(), result.Spec)
+		api_obj = &{{$node.SimpleGroupTypeName}}{{$node.BaseNodeName}}{
+			client: group.client,
+			{{$node.BaseNodeName}}: result,
+		}
+		break
+	}
+
+	if cache_obj !=nil {
+		return cache_obj,nil
+	}
+	if api_obj!=nil{
+		return api_obj,nil
 	}
```

Comment:
The api_obj should not be returned ?
It is only useful for debugging to check what is the the DB vs what is in the cached version of the same object.

Reply:
Sure @xmen4xp

Reply:
I will make the change